src/components/views/rooms/MessageComposer.js

```diff
@@ -97,18 +105,37 @@ export default class MessageComposer extends React.Component {
     }
 
     onUploadFileSelected(files) {
-        this.uploadFiles(files.target.files);
+        const tfiles = files.target.files;
+        this.uploadFiles(tfiles);
+    }
+
+    isFileUploadAllowed(file) {
+        if (this._uploadLimits.upload_size != null && file.size > this._uploadLimits.upload_size) {
```

Comment:
`!= null` is slightly confusing here since you're relying on `undefined == null` (and of course `0 == null` too, so a limit of 0 will mean unlimited). Probably better to say `=== undefined`.